### PR TITLE
[cli] Add `client.fetchPaginated()` async iterator

### PR DESCRIPTION
**Pull request by TooTallNate**
Adds async iterator helper function for interacting with endpoints that have pagination.

Implemented for use with `vc bisect` (fetching deployments) and `vc link --repo` (fetching projects).

### CODE REVIEW

File: internals/types/index.d.ts [hunk 1]
```diff
@@ -383,7 +383,7 @@ export interface ProjectLink {
 export interface PaginationOptions {
   prev: number;
   count: number;
-  next?: number;
+  next?: number | null;
```
Comment (EndangeredMassa) [inline internals/types/index.d.ts]: **question (non-blocking):** ‏Is there some type shuffling we can do to make this not both `null` and `undefined`?

**Review by EndangeredMassa — DISMISSED** [review]

File: internals/types/index.d.ts (hunk 1 above)
Comment (TooTallNate) [reply internals/types/index.d.ts]: From what I've seen so far I think the API will always return `null`. Will double check on that though.

Comment (EndangeredMassa) [reply internals/types/index.d.ts]: We could wrap it to get rid of one of the types. It's not a huge deal, but I do like simplifying these when they come up. I don't want to read the type later and wonder what the difference between `undefined` and `null` means for this property.

**Review by cb1kenobi — DISMISSED** [review]

File: internals/types/index.d.ts (hunk 1 above)
Comment (TooTallNate) [reply internals/types/index.d.ts]: Updated to use the same type as defined in API repo

**Review by EndangeredMassa — APPROVED** [review]

**Pull request merged**